packages/btc/src/api/sendRgbppUtxos.ts

```diff
@@ -0,0 +1,169 @@
+import { helpers, Hash, RawTransaction, RPC } from '@ckb-lumos/lumos';
+import { InitOutput, TxAddressOutput } from '../transaction/build';
+import { ErrorCodes, TxBuildError } from '../error';
+import { DataSource } from '../query/source';
+import { Utxo } from '../types';
+import { bitcoin } from '../bitcoin';
+import { RGBPP_UTXO_DUST_LIMIT } from '../constants';
+import { calculateCommitment } from '../ckb/commitment';
+import { unpackRgbppLockArgs } from '../ckb/molecule';
+import { getCellByOutPoint } from '../ckb/rpc';
+import { sendUtxos } from './sendUtxos';
+
+export async function sendRgbppUtxos(props: {
+  ckbVirtualTx: RawTransaction;
+  paymaster: TxAddressOutput;
+  commitment: Hash;
+  tos?: string[];
+
+  ckbNodeUrl: string;
+  rgbppLockCodeHash: Hash;
+  rgbppTimeLockCodeHash: Hash;
+  rgbppMinUtxoSatoshi?: number;
+
+  from: string;
+  source: DataSource;
+  fromPubkey?: string;
+  changeAddress?: string;
+  minUtxoSatoshi?: number;
+  feeRate?: number;
+}): Promise<bitcoin.Psbt> {
+  const inputs: Utxo[] = [];
+  const outputs: InitOutput[] = [];
+  let lastTypeInputIndex = -1;
+  let lastTypeOutputIndex = -1;
+
+  // Build TransactionSkeleton from CKB VirtualTx
+  const rpc = new RPC(props.ckbNodeUrl);
+  const ckbVirtualTx = props.ckbVirtualTx;
+  const ckbTxSkeleton = await helpers.createTransactionSkeleton(ckbVirtualTx as any, async (outPoint) => {
+    const result = await getCellByOutPoint(outPoint, rpc);
+    if (!result.cell || result.status !== 'live') {
+      throw new TxBuildError(ErrorCodes.CKB_CANNOT_FIND_OUTPOINT);
+    }
+
+    return result.cell;
+  });
+
+  // Handle and check inputs
+  const inputCells = ckbTxSkeleton.get('inputs');
+  for (let i = 0; i < inputCells.size; i++) {
+    const input = inputCells.get(i)!;
+    const isRgbppLock = input.cellOutput.lock.codeHash === props.rgbppLockCodeHash;
+    const isRgbppTimeLock = input.cellOutput.lock.codeHash === props.rgbppTimeLockCodeHash;
+
+    // If input.type !== null, input.lock must be RgbppLock or RgbppTimeLock
+    if (input.cellOutput.type) {
+      if (!isRgbppLock && !isRgbppTimeLock) {
+        throw new TxBuildError(ErrorCodes.CKB_INVALID_CELL_LOCK);
+      }
+
+      // If input.type !== null，update lastTypeInput
+      lastTypeInputIndex = i;
+    }
+
+    // If input.lock == RgbppLock, add to inputs if:
+    // 1. input.lock.args can be unpacked to RgbppLockArgs
+    // 2. utxo can be found via the DataSource.getUtxo() API
+    // 3. utxo.scriptPk == addressToScriptPk(props.from)
+    if (isRgbppLock || isRgbppTimeLock) {
```

Comment:
```suggestion
    if (isRgbppLock) {
```
no need to check `isRgbppTimeLock` in inputs